rclpy/rclpy/node.py

```diff
@@ -290,7 +291,7 @@ def get_logger(self):
     def declare_parameter(
         self,
         name: str,
-        value: ParameterValue = ParameterValue(),
+        value: Any = None,
```

Comment:
I think the type checking here could be narrower, since `Parameter()` will not all just any type, there is a limit, though the union of possible types is large.

It would be nice to improve this, but necessary in this pr.

Reply:
Edited the previous comment, meant to say `_not_ necessary`.

Reply:
Yes, I've thought about it but annotating all the types could make it a bit unreadable.
I'll have this in mind for the distro freeze.